enhancements/network/openstack-dual-stack.md

```diff
@@ -0,0 +1,249 @@
+---
+title: openstack-dual-stack
+authors:
+  - "@maysamacedo"
+reviewers:
+  - "@mandre for SME review on OpenStack and general feedback"
+  - "@cybertron for SME review on on-premise networking"
+  - "@patrickdillon for SME review and approval on installer"
+approvers:
+  - "@mandre"
+api-approvers:
+  - "None"
+creation-date: 2023-03-13
+last-updated: 2023-03-13
+tracking-link:
+  - https://issues.redhat.com/browse/OCPBU-199
+  - https://issues.redhat.com/browse/OSASINFRA-1938
+---
+
+# OpenStack dual-stack
+
+Covers adding support of OpenShift dual-stack networking on OpenStack.
+
+## Summary
+
+Customers need support for OpenShift clusters running on OpenStack with IPv4 and IPv6.
+
+## Motivation
+
+Customers need dual-stack support to allow migration of workloads from IPv4 to IPv6
+as they are running out of IPv4 addresses. This will allow OpenShift on OpenStack to
+serve a wider variety of workloads.
+
+### User Stories
+
+* As an user of a IPI OpenShift cluster, I want to have ingress and egress working with dual-stack.
+* As an user of a IPI OpenShift cluster, I want clusterIP services to be working with dual-stack.
+
+### Goals
+
+- OpenStack VMs are configured with one dual-stack interface.
+- The IPv6 Subnets can be configured with: Stateless, SLAAC or Stateful.
+- The addresses on the VM's interface are configured with the same address
+assigned by OpenStack to the port.
+- Nodes are configured with both IPv4 and IPv6 addresses, without exposing
+additional addresses, like storage interfaces.
```

Comment:
If I understand the problem correctly, this could be unrolled as:

> Each OpenShift node is configured with both IPv4 and IPv6 addresses corresponding to the cluster network; the VM does not expose additional addresses, like storage interfaces, to the OpenShift node.

Reply:
What if the OpenShift nodes _need_ access to a storage network? Does this design preclude that?

Reply:
I mentioned this as a goal to avoid the situation that is described in this issue https://github.com/kubernetes/cloud-provider/issues/56. The storage network being exposed as an internal network used by kubelet and that network not being routed within the cluster.